content/chat/rooms/message-reactions.textile

```diff
@@ -0,0 +1,192 @@
+
+```[javascript]
+    import { MessageReactionType } from '@ably/chat';
+
+    const room = await ablyChatClient.rooms.get('room1', {
+        messages: {
+            defaultMessageReactionType: MessageReactionType.Unique,
+        },
+    });
+```
+
+h2(#messages-and-reactions). Messages and reactions
+
+The @Message@ object contains a @reactions@ property which is an object that looks like this:
+
+```[javascript]
+    interface Message {
+        // ... (other fields omitted)
+        reactions: {
+            unique: Ably.SummaryUniqueValues,
+            distinct: Ably.SummaryDistinctValues,
+            multiple: Ably.SummaryMultipleValues,
+        }
+    }
+
+    // example (in real use, it is unlikely that all reaction types are present):
+    {
+        // ... other message fields omitted
+        reactions: {
+            unique: {
+                '👍': { total: 2, clientIds: ['clientA', 'clientB'] },
+                '❤️': { total: 1, clientIds: ['clientC'] },
+            },
+            distinct: {
+                '👍': { total: 2, clientIds: ['clientA', 'clientB'] },
+                '❤️': { total: 1, clientIds: ['clientA'] },
+            },
+            multiple: {
+                '👍': { total: 10, clientIds: {'clientA': 7, 'clientB': 3} },
+                '❤️': { total: 100, clientIds: {'clientA': 100} },
+            },
+        }
+    }
+```
+
+All reaction types are always available via @Message.reactions@, regardless of the default reaction type configured via room options.
+
+The @Message.reactions@ property is populated when fetching messages from history through @getPreviousMessages()@ or @room.messages.history()@. It is **not** populated when receiving message events such as @MessageEvents.Created@, @MessageEvents.Updated@, or @MessageEvents.Deleted@ from the realtime channel.
+
+Always call @Message.with(event)@ when applying message events and reaction events to existing messages to ensure that reactions are correctly copied or updated. **Do not** replace existing messages with messages received from events as reactions will be lost.
+
+
+h2(#subscribe). Subscribing to message reactions
+
+Ably generates a summary (aggregate) of the reactions for each message and for each reaction type. For displaying accurate counts for message reactions, subscribe to changes in the message summary.
+
+```[javascript]
+    room.messages.reactions.subscribe((event) => {
+        console.log("received reactions summary event", event);
+    });
+```
+
+The event is of type @reaction.summary@. @event.summary@ is the received reactions summary and contains the following properties:
+
+|_. Property |_. Description |_. Example |
+| @messageSerial@ | Serial of the message this summary is for. | @01826232498871-001&#64;abcdefghij:001@ |
+| @timestamp@ | Timestamp of the summary. | @new Date("2025-04-22T11:37:59.988Z")@ |
```

Comment:
Future consideration as not BC - should we be exposing the serial of the summary to provide ordering by the usual lexicographical method. I think probably not required, but just thinking consistency.